### PR TITLE
Add group mapping on enter

### LINKED ISSUE #16312: Enter key doesn't work to add SSO group mapping
**To Reproduce**
1. Go to Admin -> Authentication -> LDAP, SAML or JWT
2. Click on "Edit Mappings"
3. Click on "Create a Mapping"
4. Type something into the input area and hit enter

**Expected behavior**
Hitting enter after typing in a group name/DN should behave the same as clicking "Add"

Related #16047 and #2059

I'm on Firefox 88.0.1 on macOS Big Sur

### PR DESCRIPTION
**Pull request by ranquild**
Epic https://github.com/metabase/metabase/issues/19696
Fixes https://github.com/metabase/metabase/issues/16312

How to test:
- Go to Admin -> Authentication -> LDAP, SAML or JWT
- Click on "Edit Mappings"
- Click on "Create a Mapping"
- Type something into the input area and hit enter
- Check that hitting enter after typing in a group name/DN behaves the same as clicking "Add"

### CODE REVIEW

File: frontend/src/metabase/admin/settings/components/widgets/GroupMappingsWidget.jsx [hunk 1]
```diff
@@ -197,17 +197,17 @@ class AddMappingRow extends React.Component {
     };
   }
 
-  _handleCancelClick = e => {
+  _handleSubmit = e => {
     e.preventDefault();
-    const { onCancel } = this.props;
-    onCancel && onCancel();
+    const { onAdd } = this.props;
+    onAdd && onAdd(this.state.value);
```
Comment (gusaiani) [inline frontend/src/metabase/admin/settings/components/widgets/GroupMappingsWidget.jsx:203]: I guess these could be written like `onAdd?.(this.state.value)` 
¯\_(ツ)_/¯

**Review by gusaiani — APPROVED** [review]
👍

**Review by alxnddr — APPROVED** [review]
Nice! Would be great to make `esc` to trigger `cancel`

**Comment by ranquild** [comment]
> Nice! Would be great to make `esc` to trigger `cancel`

Yeah, I think `esc` should be handled on `Modal` level.

**Pull request merged**